Enable two level namespace with openmpi on MacOS in spack (release v1)

climbfuji opened the pull request:
Identical to https://github.com/NOAA-EMC/spack-stack/pull/216, but for the release v1 branch (cherry-picked the spack PR from @srherbener to make sure there are no conflicts down the road).

Waiting on https://github.com/NOAA-EMC/spack/pull/108

climbfuji committed:
Update .gitmodules and submodule pointer for spack for code review and testing

climbfuji commented:
I had problems building mapl with this PR ... not sure if it is related to the two-level namespace change or my laptop being iffy tonight. I know the CI tests look good, but I also realized that we don't build openmpi with spack in the CI tests to save time. We'll need to make the same changes in the Github actions script that builds openmpi on macOS. I can do that, but not today anymore ...

climbfuji committed:
Merge branch 'release/v1' of https://github.com/noaa-emc/spack-stack into feature/mac-openmpi-two-level-namespace-release-v1

climbfuji committed:
[skip ci] Merge branch 'release/v1' of https://github.com/noaa-emc/spack-stack into feature/mac-openmpi-two-level-namespace-release-v1

climbfuji committed:
Merge branch 'release/v1' into feature/mac-openmpi-two-level-namespace-release-v1

climbfuji committed:
Update submodule pointer for spack

climbfuji committed:
Merge branch 'release/v1' of https://github.com/noaa-emc/spack-stack into feature/mac-openmpi-two-level-namespace-release-v1

climbfuji committed:
Merge branch 'feature/mac-openmpi-two-level-namespace-release-v1' of https://github.com/climbfuji/spack-stack into feature/mac-openmpi-two-level-namespace-release-v1

srherbener reviewed (commented):
These changes look good to me. I'm testing on my iMac now.

climbfuji committed:
[skip ci] Revert .gitmodules and update submodule pointer for spack

climbfuji commented:
CI tests passed for commit https://github.com/NOAA-EMC/spack-stack/pull/217/commits/ae76960b1c6b7cc44268c0e52b59237756057aa7 . I merged the spack PR, reverted the `.gitmodules` change here and updated the spack submodule pointer.
